src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -834,6 +818,20 @@ private bool InferConnectionDetails()
                     ReportSNIError(SNIProviders.INVALID_PROV);
                     return false;
                 }
+
+                try
```

Comment:
No SSRP in Parsing logic please. You should move this out and keep it into the SNIProxy. This makes a network call, and should be separated from parsing. 
Take this decision to whether SSRP is needed or not based on the information provided by DataSource